src/menu.js

```diff
@@ -458,13 +459,32 @@ exports.get = function(props = {}) {
                 .map(syncer => syncer.id)[0]
 
             if (syncerId == null) {
-              dialog.showMessageBox(
-                i18n.t(
-                  'menu.engines',
-                  'None of the attached engines support analysis.'
-                ),
-                'info'
-              )
+              let defaultEngine = setting
+                .get('engines.list')
+                .filter(engine => engine.analysis)[0]
+
+              if (defaultEngine != null) {
+                if (!analyzeEngineInitializing) {
+                  analyzeEngineInitializing = true
+
+                  sabaki.attachEngines([defaultEngine], ({syncer, error}) => {
```

Comment:
I don't like the callback function here. `attachEngines` actually returns an array of `EngineSyncer`s that has been attached, so you can write something like:

~~~js
let [defaultEngineSyncer] = await sabaki.attachEngines([defaultEngine])

defaultEngineSyncer.controller.on('initialized', () => { /* ... */ })
~~~

Reply:
Thanks for your sample code! It is very helpful for me. Only one thing is that since the 'initialized' event may occur between sabaki.attachEngines() and defaultEngineSyncer.on('initialized', ...), I added 'autoStart' option to sabaki.attachEngines() to prevent it. However, it is very unlikely that this will happen, so I'll remove it if you don't need it.